rosbag2_transport/test/rosbag2_transport/test_play_timing.cpp

```diff
@@ -190,3 +191,43 @@ TEST_F(PlayerTestFixture, play_ignores_invalid_delay)
   EXPECT_THAT(replay_time, Gt(lower_expected_duration));
   EXPECT_THAT(replay_time, Lt(upper_expected_duration));
 }
+
+TEST_F(PlayerTestFixture, play_respects_start_offset)
+{
+  rclcpp::Duration start_delay(0, static_cast<uint32_t>(5e8));  // 5e8 ns == 0.5 s
+  rclcpp::Duration delay_margin(1, 0);
+
+  // Start 0.5 seconds into the bag
+  play_options_.start_offset = static_cast<rcutils_time_point_value_t>(RCUTILS_S_TO_NS(0.5));
+  // Subtract the delay since we've sought further into the bag
+  auto lower_expected_duration = message_time_difference - start_delay;
+  auto upper_expected_duration = message_time_difference - start_delay + delay_margin;
+  auto player = std::make_shared<rosbag2_transport::Player>(
+    std::move(reader), storage_options_, play_options_);
+
+  auto start = clock.now();
+  player->play();
+  auto replay_time = clock.now() - start;
+
+  EXPECT_THAT(replay_time, Gt(lower_expected_duration));
+  EXPECT_THAT(replay_time, Lt(upper_expected_duration));
+}
+
+TEST_F(PlayerTestFixture, play_ignores_invalid_start_offset)
+{
+  rclcpp::Duration delay_margin(1, 0);
+
+  // Player should ignore an invalid (negative) start offset
+  play_options_.start_offset = INT64_C(-5);
```

Comment:
```suggestion
  play_options_.start_offset = -5;
```

Reply:
Addressed in https://github.com/ros2/rosbag2/pull/931/commits/07543e0c770db06fbada46268105fd4eea1a9d3a.